src/firefox/index.js

```diff
@@ -239,8 +244,35 @@ export type UseProfileParams = {
   app?: PreferencesAppName,
   configureThisProfile?: ConfigureProfileFn,
   customPrefs?: FirefoxPreferences,
+  createProfileFinder?: typeof defaultCreateProfileFinder,
 };
 
+export function defaultCreateProfileFinder(userDirectoryPath?: string) {
+  const finder = new FirefoxProfile.Finder(userDirectoryPath);
+  const readProfiles = promisify(finder.readProfiles, finder);
+  return {
+    getPath: promisify(finder.getPath, finder),
+    hasProfileName: async (profileName: string) => {
+      try {
+        const profilesIniPath = path.join(
+          userDirectoryPath || FirefoxProfile.Finder.locateUserDirectory(),
+          'profiles.ini');
+
+        await fs.stat(profilesIniPath);
+        await readProfiles();
+        return finder.profiles.filter(
+          (profileDef) => profileDef.Name === profileName).length !== 0;
+      } catch (error) {
+        if (isErrorWithCode('ENOENT', error)) {
```

Comment:
This try catch should only surround the `await fs.stat` call and it should re-throw any error that does not verify `isErrorWithCode('ENOENT', error)` (or, on the contrary, this function is going to return `undefined` on all the other kind of exception that can be caught from this `try...catch`).